### DISCUSSION TITLE
Is sending audio function working?

### DISCUSSION THREAD

**Discussion by fercom** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#41] [Q&A] [answered]
I have been trying to send an audio message but have been unable to do it:

I tried like this:
` AudioMessageByUrlRequest request = new AudioMessageByUrlRequest();
                    request.Audio = new MediaAudioUrl();
                    request.Audio.Link = "https://EXAMPLEURL.mp4";
                    request.To = "521XXXXXXXXXX";
                    var result = await whatsAppBusinessClient.SendAudioAttachmentMessageByUrlAsync(request);`

And also from postman like this:
`{
    "messaging_product": "whatsapp",
    "recipient_type": "individual",
    "to": "521XXXXXXXXXX,
    "type": "audio",
    "audio": {
        "link": "https://EXAMPLEURL.mp4"
    }
}`

But it doesn't work in any way.

I cloned the code and tried to send it from the example and it doesn't work neither from ID or URL.

NOTE: I have sent a template first, and I have been able to send images and docs, it is just audio that doesn't work

**Comment by gabrieldwight** [comment, marked as answer]
Hi @fercom, yes the sending of the audio function works. I have sent using media Id by uploading an audio media to get the media Id.

**Using media Id**
![image](https://github.com/gabrieldwight/Whatsapp-Business-Cloud-Api-Net/assets/28096255/1d8dcf18-900b-4a87-96f0-379f3b8c0cbb)

**Using media Url**
![image](https://github.com/gabrieldwight/Whatsapp-Business-Cloud-Api-Net/assets/28096255/77818abf-f34d-42c6-9db8-7c981bbb6c6b)

**Note: for media URL the endpoint where you get the audio source must be public and returns the audio file metadata. It works well with cloud storage services and any endpoint URL that returns an audio file directly**

**Reply by fercom** [reply]
Thanks, I figured it out, it was that in the blob the file was saved as application/octet-stream instead of audio/ogg
